File: src/main/java/xyz/lamergameryt/allen4j/profile/SelfProfileRequest.java
package xyz.lamergameryt.allen4j.profile;

import xyz.lamergameryt.allen4j.APIEndpoint;
import xyz.lamergameryt.allen4j.Request;

public class SelfProfileRequest extends Request {
    /**
     * Set the value of the endpoint.
     */
    private SelfProfileRequest() {
        endpoint = APIEndpoint.SELF_PROFILE;
    }

    /**
     * @return An instance of this class.
     */
    public static SelfProfileRequest build() {
        return new SelfProfileRequest();
    }
}

File: src/main/java/xyz/lamergameryt/allen4j/profile/SelfProfileRequest.kt
package xyz.lamergameryt.allen4j.profile

import xyz.lamergameryt.allen4j.APIEndpoint
import xyz.lamergameryt.allen4j.Request

/**
 * The `SelfProfileRequest` class is used to perform a request that get your own profile information..
 */
class SelfProfileRequest private constructor(): Request() {
    companion object {
        /**
         * Creates an instance of this class.
         *
         * @return Instance of [SelfProfileRequest]
         */
        @JvmStatic
        fun build(): SelfProfileRequest {
            return SelfProfileRequest()
        }
    }

    /**
     * The endpoint of the [SelfProfileRequest] request.
     */
    override val endpoint = APIEndpoint.SELF_PROFILE
}